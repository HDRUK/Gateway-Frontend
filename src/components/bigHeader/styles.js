import styled from "styled-components";
import { HeaderWrapper } from "../../styles/carbonComponents.js";

export const BigHeaderWrapper = styled(HeaderWrapper)`
    display: block;
    height: 33.5rem;
    padding-top: 2rem;
`;

export const BigHeaderImage = styled.img`
    height: 6.4375rem;
    display: block;
`;
