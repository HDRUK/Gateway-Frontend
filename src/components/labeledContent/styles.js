import styled from "styled-components";

export const LabeledContentWrapper = styled.div`
    display: inline-block;

    h2,
    p {
        padding-top: 0.125rem;
        padding-right: 0.5rem;
    }
`;

export const HeaderLabel = styled.h2`
    font-size: 0.625rem;
    line-height: 0.75rem;
    color: #979797;
    text-transform: uppercase;
`;

export const ContentText = styled.p`
    font-size: 0.875rem;
    line-height: 1.0625rem;
    color: #0f62fe;

    ${p => !p.lowercase && "text-transform: uppercase"}
`;
