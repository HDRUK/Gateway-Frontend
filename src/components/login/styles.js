import styled from "styled-components";

export const LoginWrapper = styled.span`
    padding: 0 2rem;
    text-align: center;
`;
