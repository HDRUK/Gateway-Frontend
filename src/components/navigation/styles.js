import styled from "styled-components";
import { HeaderNavigation } from "carbon-components-react";

export const NavigationWrapper = styled.div`
    width: 100%;
    height: ${p => p.theme.sizes.navigation}rem;
    border-bottom: 0.0625rem solid ${p => p.theme.colors.border};
`;

export const NavigationHeader = styled(HeaderNavigation)`
    :before {
        content: unset;
    }

    @media (max-width: 66rem) {
        display: block;
    }
`;

export const NavigationItem = styled.div`
    height: 100%;
    min-height: 2.875rem;
    padding: 1rem;
    line-height: 0.875rem;
    text-decoration: none;
    color: ${p => p.theme.text.primary};

    ${p => p.active && `border-bottom: 0.25rem solid ${p.theme.colors.primarySolid}`};
    font-weight: ${p => p.active && p.theme.text.bold};

    :hover,
    :active {
        border-bottom: 0.25rem solid ${p => p.theme.colors.primarySolid};
        font-weight: ${p => p.theme.text.bold};
    }
`;
