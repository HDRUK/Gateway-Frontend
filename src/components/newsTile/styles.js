import styled from "styled-components";

export const NewsTileWrapper = styled.div`
    display: inline-block;
    width: 26rem;
    height: 27.5rem;
    overflow: hidden;
    border-radius: 0.125rem;
    box-shadow: 0 0 0.25rem 0 ${p => p.theme.colors.shadow};
`;
