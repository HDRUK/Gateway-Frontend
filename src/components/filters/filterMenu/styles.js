import styled from "styled-components";
import { AccordionBlock } from "../../../styles/carbonComponents";

export const ExpandingAccordionBlock = styled(AccordionBlock).attrs(p => ({
    style: {
        paddingBottom: p.expandheight ? `${p.expandheight}px` : "1rem"
    }
}))`
    display: inline-block;
    padding-bottom: "1rem";
`;
