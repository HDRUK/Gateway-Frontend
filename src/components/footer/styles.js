import styled from "styled-components";
import { WidthWrapper, Logo } from "../../styles/styles.js";

export const FooterWrapper = styled.div`
    height: ${p => p.theme.sizes.footer}rem;
    background: linear-gradient(135deg, ${p => p.theme.colors.primary}, ${p => p.theme.colors.secondary});
    color: white;
    padding-top: 4rem;
    padding-bottom: 1rem;
    font-size: 1.1rem;
    position: relative;
`;

export const FooterContent = styled(WidthWrapper)`
    display: block;
    margin: 0 auto 0;
    max-width: 62rem;
    padding: 0 2rem;
`;

export const FooterBlock = styled.div`
    display: inline-block;
    box-sizing: border-box;
    width: 33.33%;
    height: 100%;
    vertical-align: top;
    padding: 0 4rem 0 0;
    line-height: 1.1875rem;

    :last-of-type {
        padding-right: 0;
        padding-left: 4rem;
    }
`;

export const FooterText = styled.div`
    width: 100%;
    margin: 0 0 0.4rem 0;
`;

export const FooterImage = styled(Logo)`
    max-width: 8rem;
    max-height: 5rem;
`;
