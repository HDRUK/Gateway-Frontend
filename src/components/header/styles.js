import styled from "styled-components";
import { LinkNoDecoration } from "../../styles/styles.js";

export const HeaderAlignment = styled.div`
    width: 100%;
`;

export const HeaderAlignmentItem = styled.div`
    height: 100%;
    float: ${p => (p.right ? "right" : "left")};
`;

export const HeaderSpacer = styled.div`
    height: 100%;
    width: 6rem;
`;

export const HeaderLink = styled.a`
    :hover {
        text-decoration: underline;
        text-decoration-color: ${p => p.theme.colors.white};
    }
`;
