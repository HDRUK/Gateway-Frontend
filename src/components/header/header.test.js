import React from "react";
import { create } from "react-test-renderer";
import { MemoryRouter, Link } from "react-router-dom";
import Header from "./header.js";
import StyleWrapper from "../../HOC/StyleWrapper/StyleWrapper.js";
import { InvertedHeaderHeading, HeaderImage } from "../../styles/styles.js";
import { HeaderWrapper } from "../../styles/carbonComponents.js";
import hdr_logo_white from "../../assets/hdruk_logo_white.png";
import { AppContext } from "../../HOC/AppContext/AppContext";
import context from "../../__mocks__/AppContextMock.js";

describe("<Header> ", () => {
    let renderedComponent;
    let renderedOutput;

    beforeEach(() => {
        renderedComponent = create(
            <MemoryRouter>
                <AppContext.Provider value={context}>
                    <StyleWrapper>
                        <Header image={hdr_logo_white} headerText="Title" />
                    </StyleWrapper>
                </AppContext.Provider>
            </MemoryRouter>
        );
        renderedOutput = renderedComponent.root;
    });

    describe("when the header is rendered", () => {
        it("should render the correct components", () => {
            const wrapper = renderedOutput.findByType(HeaderWrapper);
            expect(wrapper.children[0].props["aria-label"]).toBe("HDR UK Innovation Gateway");

            const link = wrapper.findByType(Link);
            link.findByType(HeaderImage);

            const imageDestination = link.props.to;
            expect(imageDestination).toBe("/");

            const header = wrapper.findAllByType(InvertedHeaderHeading);
            expect(header[0].children[0].props.children).toBe("Innovation Gateway");
        });
    });
});
