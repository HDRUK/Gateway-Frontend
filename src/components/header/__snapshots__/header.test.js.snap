// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header>  should test the Header Snapshot 1`] = `
<header
  aria-label="HDR UK Innovation Gateway"
  className="bx--header sc-cvbbAY gKHbXx"
  role="banner"
>
  <a
    href="/"
    onClick={[Function]}
  >
    <img
      className="sc-kGXeez sc-cMljjf ktqdmc"
      src="hdruk_logo_white.png"
    />
  </a>
  <h2
    className="sc-bwzfXH sc-htpNat sc-bxivhb jNUJgX"
  >
    Innovation Gateway
  </h2>
</header>
`;
